[18.0] migration leftover product_assortment

bosd opened the pull request:
Leftover migration from #1904

CarlosRoca13 committed:
[IMP] product_assortment: Add demo data for product_assortment module

carlosdauden committed:
[FIX] product_assortment: All assortments are applied to original partner when partner is duplicated
TT48358

sergio-teruel committed:
[IMP] product_assortment: Add security group to only show assortments to managers
TT51064

bosd committed:
[IMP] product_assortment: Migration leftovers to 18.0

bosd committed:
[FIX] product_assortment: Show all applied assortments from partner smartbutton.

Before this commit, the assortments which where applied to the partner using the partner domain, where not shown.

bosd commented:
@carlosdauden Can you take a look here?

bizzappdev commented on product_assortment/models/ir_filters.py:
```diff
@@ -90,10 +88,10 @@ def get_partner_domain_fields(self):
     def _compute_all_partner_ids(self):
         """Summarize selected partners and partners from partner domain field"""
-        for ir_filter in self:
+        for ir_filter in self.sudo():
```
Not sure whether we should use compte_sudo or add self.sudo?
IMO compute_sudo is better

bosd replied:
"I’ve updated the field to use compute_sudo=True instead of adding .sudo() inside the loop. Since we are performing a search() on res.partner based on a dynamic domain, compute_sudo ensures the computation is consistent and avoids potential AccessErrors for non-admin users, while also being more performant for the Odoo cache."

bizzappdev commented on product_assortment/models/ir_rule.py:
```diff
@@ -0,0 +1,36 @@
+            "tuple(self._compute_domain_context_values())",
+        ),
+    )
+    def _compute_domain(self, model_name, mode="read"):
```
IMO, the proper place will be the search method of ir.filters

bosd replied:
Good catch! refactored it. moved this logic to the _search method of ir.filters instead. This keeps the restriction scoped to the model itself and avoids the overhead of the global rule engine

bizzappdev commented on product_assortment/models/ir_rule.py:
```diff
@@ -0,0 +1,36 @@
+        """
+        res = super()._compute_domain(model_name, mode=mode)
+        user = self.env.user
+        if model_name == "ir.filters" and not self.env.su:
```
```suggestion
        if model_name == "ir.filters" and not self.env.is_superuser():
```

bosd committed:
Update product_assortment/models/ir_rule.py

Co-authored-by: BizzAppDev <bizzappdev@users.noreply.github.com>

bosd committed:
[IMP] product_assortment: Refactor partner ID computation in ir_filters.py

bosd committed:
[REF] Restrict search for non-manager users in ir_filters

bosd commented:
please keep open.

Commit:
[IMP] product_assortment: Cover manager-only filter search

Add tests for the ir.filters _search and search_count overrides that
hide assortment filters from non-manager users, plus the count branch
of _search. These new lines were previously untested (codecov/patch).

Commit:
Merge remote-tracking branch 'origin/18.0' into 18.0-mig-product_assortment-temp

Commit:
[FIX] product_assortment: Keep admin access in restricted filter search

The _search override hid is_assortment filters from every non-manager
user. Because Odoo's ORM also calls _search to verify read access while
fetching fields (BaseModel.fetch), this denied legitimate record access
to administrators - e.g. the product_pricelist_assortment cron reading
the assortment filter raised an AccessError for a non-superuser admin
(test_cron_multicompany).

Exempt Settings administrators (base.group_system) in addition to the
superuser and the assortment manager group, consistent with the base
ir.filters security which already grants administrators access to all
filters. Regular users still cannot see assortments. Add coverage for
the admin path.

Commit:
[IMP] product_assortment: Align ir.filters search overrides with v18

Drop the vestigial `count` handling kept from the 16/17 migration: Odoo
18's `_search` has no `count` parameter and `search_count` computes the
count itself, so the branch (incl. the unreachable `except TypeError`
fallback) was dead code. Match the v18 signatures and forward the
`limit` argument that `search_count` previously dropped.
